src/psyclone/psyGen.py

```diff
@@ -1520,6 +1523,9 @@ def gen_code(self, parent):
         if not self.module_inline:
             parent.add(UseGen(parent, name=self._module_name, only=True,
                               funcnames=[self._name]))
+        else:
+            # If its inlined, the symbol must already exist
+            self.scope.symbol_table.lookup(self._name)
```

Comment:
Yes, but now that I see it I decided to catch it and reraise it as a GenerationError with relevant error message, which makes why this is here more obvious and is what gen_code errors are supposed to be.

(I initially added this because when this happened in LFRic the building system didn't fail until much later in the compilation stage, but after this PR this shouldn't be triggered anywhere in LFRic anymore :) )

Reply:
Great, I think that was what I was angling for :-)